# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.cpp
# Commit: 9ce1a2365aff
# Full Hash: 9ce1a2365afff092f9107a5f1ae1eca6a34dfce3
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-03-15 03:42:43
# Regressor Bug: 1532661
# File Overlap Count: 2
# Description:
#   Bug 1532661 - Part 4: Support initializing complete BrowsingContextGroups in a single op with an IPCInitializer struct, r=farre
#   
#   Depends on D22192
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22193
# ==============================================================================

diff -r 58f77fa35eea -r 9ce1a2365aff dom/base/nsFrameLoader.cpp
--- a/dom/base/nsFrameLoader.cpp	Thu Mar 14 18:51:07 2019 +0000
+++ b/dom/base/nsFrameLoader.cpp	Thu Mar 14 18:51:09 2019 +0000
@@ -1894,7 +1894,7 @@
     const nsAString& aName, bool aIsContent) {
   // If we're content but our parent isn't, we're going to want to start a new
   // browsing context tree.
-  if (aIsContent && !aParentContext->IsContent()) {
+  if (aIsContent && aParentContext && !aParentContext->IsContent()) {
     aParentContext = nullptr;
   }
 
@@ -2585,10 +2585,14 @@
     nsAutoString frameName;
     GetFrameName(mOwnerContent, frameName);
 
+    RefPtr<BrowsingContext> parentBC;
+    parentDocShell->GetBrowsingContext(getter_AddRefs(parentBC));
+    MOZ_ASSERT(parentBC, "docShell must have BrowsingContext");
+
     // XXX(nika): due to limitations with Browsing Context Groups and multiple
     // processes, we can't link up aParent yet! (Bug 1532661)
     RefPtr<BrowsingContext> browsingContext =
-        CreateBrowsingContext(nullptr, nullptr, frameName, true);
+        CreateBrowsingContext(parentBC, nullptr, frameName, true);
 
     mBrowserBridgeChild = BrowserBridgeChild::Create(
         this, context, NS_LITERAL_STRING(DEFAULT_REMOTE_TYPE), browsingContext);